src/quickhtml/style/css.py

```diff
@@ -0,0 +1,84 @@
+from __future__ import annotations
+from typing import Any, Generator, Union
+
+
+class StyleSheet:
+    def __init__(self, **styles):
+        self.__style_rules = styles
+
+    @property
+    def rules(self) -> dict:
+        return self.__style_rules
+
+    @classmethod
+    def from_css(cls, css: str | dict):
+        if isinstance(css, str):
+            k_v = {}
+            for statement in css.split(";"):
+                if not statement:
+                    continue
+                assert statement.count(":") == 1, f"Invalid CSS string '{css}'"
+
+                k, v = map(lambda x: x.strip(" :;\n"), css.split(":"))
+                k_v[k] = v
+
+            return cls(**k_v)
```

Comment:
I think the logic here might be a bit off. It'll parse
```py
{'color': 'red', 'font-size': '16px'}
```
And then create
```py
StyleSheet(color="red", font-size="16px")
```
Which seems to go against the intent of a `StyleSheet` since you'd want
```py
StyleSheet(h1={"color": "red", "font-size": "16px"})
```
I think it might be necessary to break this down a bit into a `StyleSheet` class and a `StyleNode` class or something similar? Where `StyleSheet` handles assigning and rendering `StyleNode`s to different elements

Reply:
Yeah I thought about this! Issue is I didn't want the end user worrying about having to create a sheet vs Node.

Currently you can do this:
```python
font_size = StyleSheet(font_size=16)
red = StyleSheet(color="red")
h1_style = Stylesheet(h1=red + font_size)
```
And it is valid.

Now there is added handling where if you tried to call `.render()` on the red (or font size) object it would raise an error.

Reply:
Hm... We can keep it like this for now. I think when we go to write docs though we should emphasize this style of creating StyleSheets
```py
style = {
    "font-size": 16,
    "color": "red",
}
h1_style = StyleSheet(h1=style)
```